src/setuptools_scm/file_finder_git.py

```diff
@@ -11,13 +11,33 @@
 
 def _git_toplevel(path):
     try:
+        cwd = os.path.abspath(path or ".")
         with open(os.devnull, "wb") as devnull:
             out = subprocess.check_output(
-                ["git", "rev-parse", "--show-toplevel"],
-                cwd=(path or "."),
+                ["git", "rev-parse", "--show-prefix"],
+                cwd=cwd,
                 universal_newlines=True,
                 stderr=devnull,
             )
+        out = out.strip()
+        if not out:
+            out = cwd
+        else:
+            cwd_parents = []
+            out_parents = ["."]
+            while True:
```

Comment:
after crosschecking with how this is handled,
i believe this can be significantly simplified

presuming the prefix is a relative path, talking the length of the prefix off the full path should yield a path ending with a path separator

the pathlib using code was a bit unfortunate

Reply:
I just try to exactly what that patch tried to do. Let me check for this soon.

Reply:
Done in https://github.com/pypa/setuptools_scm/pull/520/commits/f383bd83ea57a15e69119278a4c4df786a4a9ed3